packages/pigeon/lib/java_generator.dart

```diff
@@ -313,8 +303,37 @@ String _makeSetter(Field field) {
   return 'set$uppercased';
 }
 
-String? _javaTypeForDartType(String datatype) {
-  return _javaTypeForDartTypeMap[datatype];
+String _flattenTypeArguments(List<TypeArgument> args) {
+  return args
+      .map((TypeArgument e) => e.typeArguments == null
+          ? _javaTypeForDartTypePassthrough(e.dataType)
+          : '${_javaTypeForDartTypePassthrough(e.dataType)}<${_flattenTypeArguments(e.typeArguments!)}>')
+      .reduce((String value, String element) => '$value, $element');
+}
+
+String? _javaTypeForDartType(Field field) {
+  const Map<String, String> javaTypeForDartTypeMap = <String, String>{
+    'bool': 'Boolean',
+    'int': 'Long',
+    'String': 'String',
+    'double': 'Double',
+    'Uint8List': 'byte[]',
+    'Int32List': 'int[]',
+    'Int64List': 'long[]',
+    'Float64List': 'double[]',
+    'Map': 'Map<Object, Object>',
+  };
+  if (javaTypeForDartTypeMap.containsKey(field.dataType)) {
+    return javaTypeForDartTypeMap[field.dataType];
+  } else if (field.dataType == 'List') {
```

Comment:
Option nit (for here, 331, and 334): I find that using an `else` on an `if` whose body unconditionally returns obfuscates the early return. It's much less obvious with this construction that any code after the if/else if/else chain as written here would never execute (because there's no path that doesn't return).

Reply:
I just played around with the Dart parser to see if I could condense this to one return statement and enforce that the return value is set in each branch but Dart doesn't do that.  I filed a bug here: https://github.com/dart-lang/sdk/issues/46759

If someone added code after the if block they would get a linter error that it is dead code.